ProjectCoimbra.UWP/Project.Coimbra/Pages/LaneSettingsPage.xaml.cs

```diff
@@ -196,17 +196,29 @@ private void OptimizeKeyboard_Click(object sender, RoutedEventArgs e)
 
             this.Lanes.Children.Clear();
 
+            const int newLanes = 7;
+            this.SelectedPitchCount.Maximum = Math.Max(newLanes, this.notes.Count);
+
+            var keyboardKeys = new VirtualKey[newLanes] { VirtualKey.Number1, VirtualKey.Number2, VirtualKey.Number3, VirtualKey.Number4, VirtualKey.Number5, VirtualKey.Number6, VirtualKey.Number7 };
+            var notes = new string[newLanes] { "C", "D", "E", "F", "G", "A", "B" };
+
             var currentSymbol = '\uF146';
-            for (var currentAdd = 0; currentAdd < this.notes.Count; currentAdd++)
+            for (var currentAdd = 0; currentAdd < newLanes; currentAdd++)
             {
 #pragma warning disable CA2000 // Dispose objects before losing scope
                 var lane = new LaneSetupControl(this.AvailableSymbols, this.AvailableNotes, this.AvailableKeys);
 #pragma warning restore CA2000 // Dispose objects before losing scope
                 lane.PropertyChanged += this.Lane_PropertyChanged;
                 this.Lanes.Children.Add(lane);
-                lane.SelectedNotes.Add(this.AvailableNotes[0]);
-                lane.SelectedKeys.Add(this.AvailableKeys.First(key => string.Equals(key.Value, $"Number{(currentAdd + 1).ToString(CultureInfo.InvariantCulture)}", StringComparison.Ordinal)));
-                lane.SelectedSymbol = this.AvailableSymbols.First(symbol => string.Equals(symbol.Value, currentSymbol.ToString(CultureInfo.InvariantCulture), StringComparison.Ordinal));
+
+                foreach (var note in this.AvailableNotes.ToList().Where(note => note.Value.StartsWith(notes[currentAdd], StringComparison.InvariantCultureIgnoreCase)))
```

Comment:
I think you can likely skip the `.ToList()` here as you're only enumerating the through the list once